cdap-client/src/main/java/co/cask/cdap/client/ProgramClient.java

```diff
@@ -397,6 +401,38 @@ public String call() throws Exception {
       Throwables.propagateIfPossible(e.getCause(), UnauthenticatedException.class);
       Throwables.propagateIfPossible(e.getCause(), ProgramNotFoundException.class);
       Throwables.propagateIfPossible(e.getCause(), IOException.class);
+      Throwables.propagate(e.getCause());
+    }
+  }
+
+  /**
+   * Waits for a program to have a certain status.
+   *
+   * @param program the program
+   * @param status the desired status
+   * @param timeout how long to wait in milliseconds until timing out
+   * @throws IOException if a network error occurred
+   * @throws ProgramNotFoundException if the program with the specified name could not be found
+   * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
+   * @throws TimeoutException if the program did not achieve the desired program status before the timeout
+   * @throws InterruptedException if interrupted while waiting for the desired program status
+   */
+  public void waitForStatus(final ProgramId program, String status, long timeout, TimeUnit timeoutUnit)
```

Comment:
same here - should not add a duplicate method. The older method should be deprecated and delegate to this one.